src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -115,7 +115,7 @@ public IEnumerable<User> Users
                     return new List<User>();
                 }
 
-                return UserTokenCache.GetUsers(ClientId);
+                return UserTokenCache.GetUsers(new Uri(Authority).Host);
```

Comment:
see the bug filed in your repo.

---
In reply to: [109343735](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/282#discussion_r109343735) [](ancestors = 109343735)